### PR TITLE
Add spare2 cheaper algorithm for larger workload

### PR DESCRIPTION
**Pull request by methane**

### CODE REVIEW

File: uwsgi.h [hunk 1]
```diff
@@ -1841,6 +1841,8 @@ struct uwsgi_server {
 	// minimal number of running workers in cheaper mode
 	int cheaper_count;
 	int cheaper_initial;
+	// how slowly decrease workers
+	int cheaper_spare2_idlecount;
```
Comment (unbit) [inline uwsgi.h]: place this at the end of the struct to maintain ABI compatibility

File: core/uwsgi.c [hunk 2]
```diff
@@ -790,6 +790,7 @@ static struct uwsgi_option uwsgi_base_options[] = {
 	{"cheaper-algo", required_argument, 0, "choose to algorithm used for adaptive process spawning", uwsgi_opt_set_str, &uwsgi.requested_cheaper_algo, UWSGI_OPT_MASTER},
 	{"cheaper-step", required_argument, 0, "number of additional processes to spawn at each overload", uwsgi_opt_set_int, &uwsgi.cheaper_step, UWSGI_OPT_MASTER | UWSGI_OPT_CHEAPER},
 	{"cheaper-overload", required_argument, 0, "increase workers after specified overload", uwsgi_opt_set_64bit, &uwsgi.cheaper_overload, UWSGI_OPT_MASTER | UWSGI_OPT_CHEAPER},
+	{"cheaper-spare2-idlecount", required_argument, 0, "number of secs to decrease workers when idle", uwsgi_opt_set_int, &uwsgi.cheaper_spare2_idlecount, UWSGI_OPT_MASTER | UWSGI_OPT_CHEAPER},
```
Comment (unbit) [inline core/uwsgi.c]: just call this cheaper-idle so other algorithms (in the future) could use it. Check the help line too, it looks weird to me, is not it the number of workers to remove (basically the opposite of cheaper-step)

Comment (methane) [inline core/uwsgi.c]: How do I do about spare algorithm?
It uses `cheaper-overload` (default: 3) to wait decreasing workers.

* spare algorithm uses `cheaper-idle` too.  It's 3 by default to keep compatibility to default of `cheaper-overload`.
* don't touch spare algorithm. Add `(spare2)` in option document.

Comment (unbit) [inline core/uwsgi.c]: the second one :)

Comment (methane) [inline core/uwsgi.c]: I did it.

**Comment by unbit** [comment]
Thanks

**Comment by methane** [comment]
FYI, I've created plugin for uWSGI 2.0
https://github.com/KLab/uwsgi-cheaper-spare2

**Pull request merged**